ui/dashboards/src/context/DatasourceStoreProvider.tsx

```diff
@@ -49,22 +49,11 @@ export type BuildDatasourceProxyUrlParams = {
 
 export type BuildDatasourceProxyUrlFunc = (p: BuildDatasourceProxyUrlParams) => string;
 
-/**
- * The external API for fetching datasource resources
- */
-export interface DatasourceApi {
-  buildProxyUrl?: BuildDatasourceProxyUrlFunc;
-  getDatasource: (project: string, selector: DatasourceSelector) => Promise<DatasourceResource | undefined>;
-  getGlobalDatasource: (selector: DatasourceSelector) => Promise<GlobalDatasourceResource | undefined>;
-  listDatasources: (project: string, pluginKind?: string) => Promise<DatasourceResource[]>;
-  listGlobalDatasources: (pluginKind?: string) => Promise<GlobalDatasourceResource[]>;
-}
-
 /**
  * A `DatasourceContext` provider that uses an external API to resolve datasource selectors.
  */
 export function DatasourceStoreProvider(props: DatasourceStoreProviderProps): ReactElement {
-  const { projectName, datasourceApi, onCreate, children } = props;
+  const { projectName, datasources, buildProxyUrl, onCreate, children } = props;
```

Comment:
Why did we change the datasourceApi injection into a collection of datasources? The API allows to query the datasources lazily and uses the filtering on request rather than having to load all the datasources and do the filtering in the frontend.